end_to_end_tests/openapi.json

```diff
@@ -396,6 +396,14 @@
             },
             "name": "enum_prop",
             "in": "query"
+          },
+          {
+            "required": false,
+            "schema": {
+              "$ref": "#/components/schemas/ModelWithUnionProperty"
+            },
+            "name": "model_prop",
+            "in": "query"
```

Comment:
You should probably also add `nullable` and `not required and nullable` examples as query params just to verify the template is working in all those cases using e2e tests.

Reply:
@dbanty Is it OK if I add that test coverage in https://github.com/triaxtec/openapi-python-client/pull/332? Since it's not possible AFAIK to make a nullable model query parameter without making it a union type, and if I make it a union type then it depends on https://github.com/triaxtec/openapi-python-client/pull/332.

Reply:
Yep that's fine.